export/mlab_export.py

```diff
@@ -0,0 +1,591 @@
+        options.rrddir_prefix and all option.show_* flags.
+  Returns:
+    (str, str, str), with HOSTNAME, experiment name, metric name.
+  """
+  # Strip rrddir_prefix and split directory components.
+  file_fields = filename.replace(options.rrddir_prefix, '').split('/')
+
+  # The zeroth field is always the context hostname.
+  if HOSTNAME == file_fields[0]:
+    # The root context represents whole-system metrics.
+    experiment = 'system'
+  else:
+    # A slice hostname. Everything remaining after stripping hostname.
+    experiment = file_fields[0].replace('.' + HOSTNAME, '')
+
+  raw_fields = file_fields[1:-1]
+  raw_fields.append(file_fields[-1].replace('.rrd', ''))
+  if value_name != 'value':
+    raw_fields.append(value_name)
+
+  metric_raw = '.'.join(raw_fields)
+
+  # NOTE: convert the raw metric name into the canonical form, or None.
+  metric = METRIC_MAP.get(metric_raw, None)
+
+  # Optionally print extra information.
+  if options.show_nagios:
+    cmd = ('collectd-nagios -s $COLLECTD_UNIXSOCK -H {host} '+
+           '-n {metric} -d {value} [-w <l:h> -c <l:h>]')
+    cmd = cmd.format(host=file_fields[0],
+                     metric='/'.join(file_fields[1:]).replace('.rrd', ''),
+                     value=value_name)
+    logging.info(cmd)
+  if options.show_rrdfile:
+    logging.info("rrdfile: %s", filename.replace(options.rrddir_prefix, ''))
+  if options.show_rawmetric:
+    logging.info("metric_raw: %s", metric_raw)
+  if options.show_skipped and not metric:
+    logging.info("metric_skipped: %s", metric_raw)
+  if options.show_metric and metric:
+    logging.info("metric: %s", metric)
+
+  return (HOSTNAME, experiment, metric)
+
+
+def get_json_record(hostname, experiment, metric, timestamps, values):
+  """Creates a dict suitable for export to json.
+
+  Args:
+    hostname, str, hostname of host system.
+    experiment: str, name of experiment running on host.
+    metric: str, the canonical metric name for values.
+    timestamps: iterable of int, timestamps corresponding to each value.
+    values: iterable of float, values corresponding to each timestamp.
+  Returns:
+    dict, with keys for hostname, experiment, metric, and sample.
+  """
+  logging.debug('%s %s %s', hostname, experiment, metric)
+  json_data = {
+      'hostname': hostname, 'experiment': experiment, 'metric': metric}
+  json_data['sample'] = get_json_record_samples(timestamps, values)
+  return json_data
+
+
+def get_json_record_samples(timestamps, values):
+  """Converts a sequences of timestampes and values for a json record.
+
+  The timestamps and values arguments must be the same length.
+
+  Args:
+    timestamps: iterable of int, timestamps corresponding to each value.
+    values: iterable of float, values corresponding to each timestamp.
+  Returns:
+    list of dict, each dict has keys timestamp and value.
+  """
+  samples = []
+  assert(len(timestamps) == len(values))
+  for i in xrange(0, len(timestamps)):
```

Comment:
Oh, I did notice that the initial `0` here is unnecessary for `range` and `xrange`. bc8ea7274f39e223b0199a280b1f9ec3ab236095